code/modules/events/supermatter_surge.dm

```diff
@@ -0,0 +1,23 @@
+/datum/round_event_control/supermatter_surge
+	name = "Supermatter Surge"
+	typepath = /datum/round_event/supermatter_surge
+	weight = 20
+	max_occurrences = 4
+	earliest_start = 10 MINUTES
+
+/datum/round_event_control/supermatter_surge/canSpawnEvent()
+	if(GLOB.main_supermatter_engine?.has_been_powered)
+		return ..()
+
+/datum/round_event/supermatter_surge
+	var/power = 2000
+
```

Comment:
You should add an `announceWhen = X` here. Mostly because it lets admins choose whether the announcement goes out if they trigger the event, which most events like this have. It should just be a non-0 to work IIRC.